src/cluster.c

```diff
@@ -1820,6 +1862,102 @@ void clusterUpdateSlotsConfigWith(clusterNode *sender, uint64_t senderConfigEpoc
     }
 }
 
+/* Cluster ping extensions.
+ *
+ * The ping/pong/meet messages support arbitrary extensions to add additional
+ * metadata to the messages that are sent between the various nodes in the
+ * cluster. The extensions take the form:
+ * [ Header length + type (8 bytes) ] 
+ * [ Extension information (Arbitrary length, but must be 8 byte padded) ]
+ */
+
+
+/* Returns the length of a given extension */
+static uint16_t getPingExtLength(clusterMsgPingExt *ext) {
```

Comment:
Looks like `uint16_t` is wrong here, `length` is `uint32_t`. Given the flood of int overflows we've encountered I think we should try to stick to `size_t` internally.